x-pack/legacy/plugins/maps/public/layers/sources/es_search_source/es_search_source.js

```diff
@@ -219,33 +219,43 @@ export class ESSearchSource extends AbstractESSource {
     ];
   }
 
-  async _excludeDateFields(fieldNames) {
-    const dateFieldNames = (await this.getDateFields()).map(field => field.getName());
-    return fieldNames.filter(field => {
-      return !dateFieldNames.includes(field);
+  _getDocValueAndSourceFields(indexPattern, fieldNames) {
```

Comment:
there is no reference to `this`, so this isn't a method. I would  extract this to a utility function.

When extracted, we could add unit-test with some mocked index-pattern and field-list. We can test the expected behavior with returns to splitting this up in 3 collection (scripted, sourceonly, docvalues).

btw. this function reads a lot better than before.